src/XRP/xrp-payment.ts

```diff
@@ -0,0 +1,79 @@
+import { Payment } from '../generated/web/org/xrpl/rpc/v1/transaction_pb'
+import XRPCurrencyAmount from './xrp-currency-amount'
+import XRPPath from './xrp-path'
+
+// TODO(amiecorso): Modify this object to use X-Address format.
+export default class XRPPayment {
+  public static from(payment: Payment): XRPPayment | undefined {
+    let amount
+    const paymentAmountValue = payment.getAmount()?.getValue()
+    if (paymentAmountValue) {
+      amount = XRPCurrencyAmount.from(paymentAmountValue)
+      if (!amount) {
+        return undefined // amount is required
+      }
+    } else {
+      return undefined
+    }
+    const destination = payment
+      .getDestination()
+      ?.getValue()
+      ?.getAddress()
+    if (!destination) {
+      return undefined // destination is required
+    }
+    const destinationTag = payment.getDestinationTag()?.getValue()
+    // If the deliverMin field is set, it must be able to be transformed into a XRPCurrencyAmount.
```

Comment:
nit: blank line to indicate that `destinationTag` is not related to `deliverMin`